### ISSUE TITLE
K3s coreDNS dial tcp: lookup: server misbehaving during HTTP-01 challenge

### ISSUE THREAD

**Issue by Chris2301** [k3s-io/k3s#6948]
During the apply of my first app of domain example1.nl I have added LetsEncrypt HTTP-01 succesfully

However now when trying an app domain example2.nl I get the following error (so a second domain to the cluster)

`Waiting for HTTP-01 challenge propagation: failed to perform self check GET request 'http://example2.nl/.well-known/acme-challenge/9kdpAMRFKpt_t8SaCB4fM8itLesLxPkgT58RNeRCwww': Get "http://example2.nl/.well-known/acme-challenge/9kdpAMRFKpt_t8SaCB4fM8itLesLxPkgT58RNeRCwww": dial tcp: lookup example2.nl on 10.43.0.11:53: server misbehaving`

I do not understand what has changed with the second service. When CoreDNS had trouble and needed specific config, I would have guessed it already had errors when trying the first app? 

The setup is a k3s single node cluster on a VPS and I added the domains to the /etc/hosts file on de VPS. Domains are  both pointing to the VPS. 

Can someone explain me what to investigate further? To my knowledge I have traefik ingresses to point to the right services internally and I am a bit lost what coreDNS does in this. So a pointer to the right documentation could also be helpful.

If you need config files or yamls please ask! Ty in advance!

**Comment by rbrtbnfgl** [comment]
If you add the domains on `/etc/hosts/` of the nodes does it work?

**Comment by brandond** [comment]
> The setup is a k3s single node cluster on a VPS and I added the domains to the /etc/hosts file on de VPS. Domains are both pointing to the VPS.

I'm confused by what you're trying to accomplish here. If the domains exist, and have records pointed at this node, why do you need to put them in your hosts file?

**Comment by Chris2301** [comment]
@brandond & @rbrtbnfgl thanks for you questions and I can understand de confusion, I was confused as well. I interpreted the 'server misbehaving' as an internal problem because I saw the ip of the CoreDNS. 

After a check of the DNS I saw an old NS in my records that I already deleted and that made me thinking of something my Registrar had done for me a year ago at the registry. I asked them to look at it and they switched the NS's back to what I had configured, making the DNS resolve again. 

Sorry for the confusion, that IP really mad me think internally only.
